hare/hare.go

```diff
@@ -342,14 +344,43 @@ func (h *Hare) onTick(ctx context.Context, id types.LayerID) (bool, error) {
 		mchOut: h.mchMalfeasance,
 		report: h.outputChan,
 	}
-	cp := h.factory(h.ctx, h.config, instID, set, h.rolacle, h.sign, h.publisher, comm, clock)
+	cp := h.factory(h.ctx, h.config, lid, set, h.rolacle, h.sign, h.publisher, comm, clock)
 	cp.Start()
-	h.patrol.SetHareInCharge(instID)
-	logger.With().Debug("number of consensus processes (after register)",
-		log.Int32("count", atomic.AddInt32(&h.totalCPs, 1)))
+	h.addCP(logger, cp)
+	h.patrol.SetHareInCharge(lid)
 	return true, nil
 }
 
+func (h *Hare) addCP(logger log.Log, cp Consensus) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.cps[cp.ID()] = cp
+	logger.With().Info("number of consensus processes (after register)",
+		log.Int("count", len(h.cps)))
+}
+
+func (h *Hare) getCP(lid types.LayerID) Consensus {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	return h.cps[lid]
```

Comment:
It could benefit from `RWMutex` too if the `value, ok := h.cps[lid]` notation was used.